metals/src/main/scala/scala/meta/internal/builds/NewProjectProvider.scala

```diff
@@ -0,0 +1,358 @@
+package scala.meta.internal.builds
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.Try
+
+import scala.meta.internal.metals.ClientCommands
+import scala.meta.internal.metals.ClientConfiguration
+import scala.meta.internal.metals.Icons
+import scala.meta.internal.metals.JsonParser._
+import scala.meta.internal.metals.Messages._
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.MetalsInputBoxParams
+import scala.meta.internal.metals.MetalsLanguageClient
+import scala.meta.internal.metals.MetalsOpenWindowParams
+import scala.meta.internal.metals.MetalsQuickPickItem
+import scala.meta.internal.metals.MetalsQuickPickParams
+import scala.meta.internal.metals.StatusBar
+import scala.meta.internal.metals.Time
+import scala.meta.io.AbsolutePath
+
+import coursierapi._
+import org.eclipse.lsp4j.ExecuteCommandParams
+
+class NewProjectProvider(
+    buildTools: BuildTools,
+    client: MetalsLanguageClient,
+    statusBar: StatusBar,
+    config: ClientConfiguration,
+    time: Time,
+    shell: ShellRunner,
+    icons: Icons,
+    workspace: AbsolutePath
+)(implicit context: ExecutionContext) {
+
+  private val templatesUrl =
+    "https://github.com/foundweekends/giter8/wiki/giter8-templates.md"
+  private val gitterDependency = Dependency
+    .of("org.foundweekends.giter8", "giter8_2.12", "0.12.0")
+  // equal to cmd's: g8 playframework/play-scala-seed.g8 --name=../<<name>>
+  private val gitterMain = "giter8.Giter8"
+
+  private val allTemplates = mutable.Seq.empty[MetalsQuickPickItem]
```

Comment:
this never mutates, it doesn't need to be mutable. 
Then somewhere below you don't need to call `.toSeq`

Reply:
It does mutate, we update the templates in it.

Reply:
noo wait, where do you set it?

Reply:
och, actually forgot about that one. Now it downloads each time. uff...

Reply:
Now it should be good.